src/common/cryptoutils.cpp

```diff
@@ -386,8 +386,21 @@ Error ParsePKCS11URL(
         return AOS_ERROR_WRAP(err);
     }
 
-    err = FindUrlParam(url, "pin-value", userPin);
-    if (!err.IsNone() && !err.Is(ErrorEnum::eNotFound)) {
+    StaticString<cFilePathLen> pinPath;
+
+    err = FindUrlParam(url, "pin-source", pinPath);
+    if (err.Is(ErrorEnum::eNotFound)) {
+        userPin.Clear();
```

Comment:
why should user pin be cleared?
looks like it's empty at this stage.

Reply:
it depends on the caller whether userPin empty or not. this check ensures, that it is indeed empty if pin-source is not provided.